grid-client/deployer/deployment_deployer.go

```diff
@@ -28,9 +29,9 @@ func NewDeploymentDeployer(tfPluginClient *TFPluginClient) DeploymentDeployer {
 }
 
 // GenerateVersionlessDeployments generates a new deployment without a version
-func (d *DeploymentDeployer) GenerateVersionlessDeployments(ctx context.Context, dl *workloads.Deployment) (map[uint32]gridtypes.Deployment, error) {
+func (d *DeploymentDeployer) GenerateVersionlessDeployments(ctx context.Context, dl *workloads.Deployment, usedHosts *[]byte) (map[uint32]gridtypes.Deployment, error) {
```

Comment:
I was wondering why you passing a `pointer to an array` but i realized you do this to `modify` the array. So it works as an output.

But why not return the array instead of using this output pattern ?

Reply:
I am also wondering what a `[]byte` represents here. there are no docs on what the inputs are? not the type is descriptive (no type alias with a descriptive name)?

Reply:
It's the last byte in the IP so for 1.1.1.2 we just store 2, I personally do not find the best way to store that but this is how it's used throughout the grid client. We were considering changing it to something like BitSet but I believe this will need some refactoring outside this pr scope